inventory_management_system_api/services/utils.py

```diff
@@ -184,22 +184,20 @@ def _check_missing_mandatory_catalogue_item_properties(
             )
 
 
-def _filter_matching_catalogue_item_properties(
-    defined_properties: Dict[str, Dict],
-    supplied_properties: Dict[str, Dict],
+def _merge_non_mandatory_catalogue_item_properties(
+    defined_properties: Dict[str, Dict], supplied_properties: Dict[str, Dict]
 ) -> Dict[str, Dict]:
     """
-    Filter through the supplied properties and extract the ones matching the defined properties.
-
-    :param defined_properties: The defined catalogue item properties stored as part of the catalogue category in the
-        database.
-    :param supplied_properties: The supplied catalogue item properties.
-    :return: The supplied properties that are matching the defined properties.
+    Merges in any non-mandatory properties that have not been supplied, giving them a value of None, using
+    the same order as they are defined.
     """
-    logger.info("Extracting the supplied properties that are matching the defined properties")
-    matching_properties = {}
-    for supplied_property_name, supplied_property in supplied_properties.items():
-        if supplied_property_name in defined_properties:
-            matching_properties[supplied_property_name] = supplied_property
+    logger.info("Merging any missing defined non-mandatory properties with the supplied properties")
 
-    return matching_properties
+    properties: Dict[str, Dict] = {}
+    for defined_property_name, defined_property in defined_properties.items():
+        supplied_property = supplied_properties.get(defined_property_name)
```

Comment:
Does `get` return `None` if it cannot find the provided key even if `None` isn't provided as the default return value?